kii_thing_if.h

```diff
@@ -379,6 +394,90 @@ kii_bool_t init_kii_thing_if_with_onboarded_thing(
         kii_thing_if_state_updater_resource_t* state_updater_resource,
         KII_JSON_RESOURCE_CB resource_cb);
 
+/** Upate firmware version of a thing.
+ *
+ * This function must be called between ::activate and one of
+ * following functions:
+ * - ::onboard_with_vendor_thing_id,
+ * - ::onboard_with_thing_id
+ * - ::init_kii_thing_if_with_onboarded_thing
+ *
+ * @param [in] kii_thing_if_t This SDK instance.
+ * @param [in] firmware_version firmware version to update.
+ * @return KII_TRUE when succeeded, KII_FALSE when failed.
+ */
+kii_bool_t update_firmware_version(
+        kii_thing_if_t* kii_thing_if,
+        const char* firmware_version);
+
+/** Get current firmware version of a thing.
+ *
+ * This function must be called between ::activate and one of
+ * following functions:
+ * - ::onboard_with_vendor_thing_id,
+ * - ::onboard_with_thing_id
+ * - ::init_kii_thing_if_with_onboarded_thing
+ *
+ * Applications <b> must not free </b> the returned value.
+ *
+ * The returned value is valid until other API is called. After other
+ * API is called, the returned value might be broken. If you want to
+ * use the returned value over other API call, you need to copy the
+ * returned value in your own.
+ *
+ * @param [in] kii_thing_if_t This SDK instance.
+ * @return current firmware version. Contents of the return value is
+ * one of the followings:
+ * - If succeed to get current firmware version, this function
+ *   returnes null terminated string denoting firmware version.
+ * - If thing does not have firmware version, this function returns
+ *   empty string. This string also null terminated.
+ * - Otherwise, returns NULL.
+ */
+const char* get_current_firmware_version(kii_thing_if_t* kii_thing_if);
```

Comment:
I think there's downside of returning char*.

- "" / NULL can be used to distinguish the error but probably not very explicit.
- Memory is allocated in somewhere by SDK. (Probably static.)  It is not very clear that developer has to free or not.